### ISSUE TITLE
dired-annotator-show-icons returns error

### ISSUE THREAD

**Issue by ouboub** [gunther-bachmann/emacs-dired-annotator#3]
Hi

with the standard setting 
```
(use-package dired-annotator
  :load-path "~/ALLES/src/emacs-dired-annotator"
  :bind (:map dired-mode-map
              ("C-c a s" . #'dired-annotator-show-icons)
              ("C-c a h" . #'dired-annotator-hide-icons)
              ("C-c a k" . #'dired-annotator-delete-note)
              ("C-c a o" . #'dired-annotator-edit-note)
              ("C-c a e" . #'dired-annotator-edit-note)
              ("C-c a a" . #'dired-annotator-show-note)
              ("'" . #'dired-annotator-show-note)))
```
running dired-annotator-show-icons
returns, independently where the cursor is:

```
Debugger entered--Lisp error: (error "Cannot operate on ‘.’ or ‘..’")
  signal(error ("Cannot operate on ‘.’ or ‘..’"))
  error("Cannot operate on `.' or `..'")
  dired-get-filename()
  dired-annotator--show-icons-in-region(1 632)
  dired-annotator--show-icons()
  dired-annotator-show-icons()
  funcall-interactively(dired-annotator-show-icons)
  call-interactively(dired-annotator-show-icons nil nil)
  command-execute(dired-annotator-show-icons)

```

**Comment by gunther-bachmann** [comment]
wow. never seen this. guess I will make sure that dired-get-filename is not executed on these 'files'

**Comment by gunther-bachmann** [comment]
Could you please retry with current head. I added an (ignore-errors ...) around the call to dired-get-filename, which should ignore illegal positions that make dired-get-filename fail.

**Comment by ouboub** [comment]
>>> "GB" == Gunther Bachmann ***@***.***> writes:

> Could you please retry with current head. I added an (ignore-errors
> ...) around the call to dired-get-filename, which should ignore
> illegal positions that make dired-get-filename fail.
Ok

-- 
Warning: Content may be disturbing to some audiences
I strongly condemn Putin's war of aggression against the Ukraine.
I support to deliver weapons to Ukraine's military. 
I support the ban of Russia from SWIFT.
I support the EU membership of the Ukraine. 
https://addons.thunderbird.net/en-US/thunderbird/addon/gmail-conversation-view/

**Comment by ouboub** [comment]
> Could you please retry with current head. I added an (ignore-errors ...) around the call to dired-get-filename, which should ignore illegal positions that make dired-get-filename fail.

I pulled the current HEAD, and recompiled: the errors are now
```
In end of data:
dired-annotator.el:704:54: Warning: the function
    ‘dired-annotator--subtree--cleanup-icons-after-fold’ is not known to be
    defined.
dired-annotator.el:701:54: Warning: the function
    ‘dired-annotator--subtree--possibly-show-for-inserted’ is not known to be
    defined.
dired-annotator.el:634:103: Warning: the function
    ‘dired-annotator--get-all-tags’ is not known to be defined.
dired-annotator.el:634:6: Warning: the function
    ‘dired-annotator--minimal-narrow’ is not known to be defined.
dired-annotator.el:502:21: Warning: the function
    ‘dired-annotator--get-tags-for’ is not known to be defined.
```

besides the init conf:
```
(use-package dired-annotator
  :after (posframe all-the-icons org-ql dired-narrow)
  :load-path "~/ALLES/src/emacs-dired-annotator"
  :custom ((dired-annotator-note-icon (all-the-icons-faicon "sticky-note"))
           (dired-annotator-annotations-folder (expand-file-name "~/documents/annotations"))
           (dired-annotator-note-popup-hook #'dired-annotator--posframe-note-popup)
           (dired-annotator-note-fill-column 54) ;; a bit smaller than the popup window
           (dired-annotator-note-popup-remove-hook '(dired-annotator-register-buffer-cleanup
                                                     dired-annotator--remove-all-posframes)))
  :hook (dired-mode . dired-annotator-check-dir-local-show)
  :init
  (defconst dired-annotator-note-popup-width 56 "width of the note that pops up")
  (defconst dired-annotator-note-popup-height 40 "max height of the note that pops up")

  (defun dired-annotator--remove-all-posframes ()
    "remove all posframes"
    (unless (eq 'dired-annotator-show-note (lookup-key dired-mode-map (this-single-command-keys)))
      (setq dired-annotator--note-should-not-popup nil))
    (ignore-errors
      (posframe-hide-all)))

  (defun dired-annotator--posframe-note-popup (position annotation-file)
    (setq dired-annotator--note-should-not-popup t)
    (let ((note-buffer (find-file-noselect annotation-file)))
      (with-current-buffer note-buffer
        ;; make sure that regardless of org startup visibility, the buffer is fully visible
        (org-show-all))
      (posframe-show note-buffer :position position :lines-truncate t
                     :width dired-annotator-note-popup-width
                     :border-color "plum4" :border-width 2 :left-fringe 3 :right-fringe 3 )))

  :bind (:map dired-mode-map
              ("C-c a r" . #'dired-annotator-reload-annotation-info)
              ("C-c a d" . #'dired-annotator-dired-on-annotation-folder)
              ("C-c a s" . #'dired-annotator-show-icons)
              ("C-c a h" . #'dired-annotator-hide-icons)
              ("C-c a k" . #'dired-annotator-delete-note)
              ("C-c a o" . #'dired-annotator-edit-note)
              ("C-c a e" . #'dired-annotator-edit-note)
              ("C-c a a" . #'dired-annotator-show-note)
              ("C-c a /" . #'dired-annotator-narrow-on-tag)
              ("'" . #'dired-annotator-show-note)))
```
still does not work
none of the dired-annoator-functions are found.

I am now really puzzled
however the simple setting

```
(use-package dired-annotator
  :after (posframe all-the-icons org-ql dired-narrow)
  :load-path "~/ALLES/src/emacs-dired-annotator"
  :custom ((dired-annotator-note-icon (all-the-icons-faicon "sticky-note"))
           (dired-annotator-annotations-folder (expand-file-name "~/documents/annotations"))
           (dired-annotator-note-popup-hook #'dired-annotator--posframe-note-popup)
           (dired-annotator-note-fill-column 54) ;; a bit smaller than the popup window
           (dired-annotator-note-popup-remove-hook '(dired-annotator-register-buffer-cleanup
                                                     dired-annotator--remove-all-posframes)))
  :hook (dired-mode . dired-annotator-check-dir-local-show)
  :init
  (defconst dired-annotator-note-popup-width 56 "width of the note that pops up")
  (defconst dired-annotator-note-popup-height 40 "max height of the note that pops up")

  (defun dired-annotator--remove-all-posframes ()
    "remove all posframes"
    (unless (eq 'dired-annotator-show-note (lookup-key dired-mode-map (this-single-command-keys)))
      (setq dired-annotator--note-should-not-popup nil))
    (ignore-errors
      (posframe-hide-all)))

  (defun dired-annotator--posframe-note-popup (position annotation-file)
    (setq dired-annotator--note-should-not-popup t)
    (let ((note-buffer (find-file-noselect annotation-file)))
      (with-current-buffer note-buffer
        ;; make sure that regardless of org startup visibility, the buffer is fully visible
        (org-show-all))
      (posframe-show note-buffer :position position :lines-truncate t
                     :width dired-annotator-note-popup-width
                     :border-color "plum4" :border-width 2 :left-fringe 3 :right-fringe 3 )))

  :bind (:map dired-mode-map
              ("C-c a r" . #'dired-annotator-reload-annotation-info)
              ("C-c a d" . #'dired-annotator-dired-on-annotation-folder)
              ("C-c a s" . #'dired-annotator-show-icons)
              ("C-c a h" . #'dired-annotator-hide-icons)
              ("C-c a k" . #'dired-annotator-delete-note)
              ("C-c a o" . #'dired-annotator-edit-note)
              ("C-c a e" . #'dired-annotator-edit-note)
              ("C-c a a" . #'dired-annotator-show-note)
              ("C-c a /" . #'dired-annotator-narrow-on-tag)
              ("'" . #'dired-annotator-show-note)))
```
works so something is wrong with the complex setting.

**Comment by ouboub** [comment]
> wow. never seen this. guess I will make sure that dired-get-filename is not executed on these 'files'

may be the reason is the incorrect file entry 
#+property: file-information ("Unmatched '.
Unmatched '.
" "/home/oub/ALLES/emacs/init/" "emacs_init.el" 80791 "2022-11-02T20:43:37" immutable-file)

**Comment by gunther-bachmann** [comment]
This annotation is definitely 'broken'. could you replace the props line please with `#+property: file-information ("6e7d1006392baeab770d3d8f1e34a089" "/home/oub/ALLES/emacs/init/" "emacs_init.el" 80791 "2022-11-02T20:43:37" immutable-file)`
This should make this annotation 'functional' even though the hash is not matching the file. Try creating new annotations with the current HEAD of the repo. I hope the md5 call is now returning the correct hash for files on your machine. If not, I'll add some logging that might help.

**Comment by ouboub** [comment]
>>> "GB" == Gunther Bachmann ***@***.***> writes:

> This annotation is definitely 'broken'. could you replace the props
> line please with `#+property: file-information
> ("6e7d1006392baeab770d3d8f1e34a089" "/home/oub/ALLES/emacs/init/"
> "emacs_init.el" 80791 "2022-11-02T20:43:37" immutable-file)`
> This should make this annotation 'functional' even though the hash is
> not matching the file. Try creating new annotations with the current
> HEAD of the repo. I hope the md5 call is now returning the correct
> hash for files on your machine. If not, I'll add some logging that
> might help.

I tried this already and it did not help. If you don't mind I will write
you an email (in German I would say) to your address that figures in the
git repository, because I have now quite a bit of issues, some of them
are overlapping. I also think it is better to concentrate first on the
simpler setting


```
(use-package dired-annotator
  :load-path "~/ALLES/src/emacs-dired-annotator"
  :bind (:map dired-mode-map
              ("C-c a s" . #'dired-annotator-show-icons)
              ("C-c a h" . #'dired-annotator-hide-icons)
              ("C-c a k" . #'dired-annotator-delete-note)
              ("C-c a o" . #'dired-annotator-edit-note)
              ("C-c a e" . #'dired-annotator-edit-note)
              ("C-c a a" . #'dired-annotator-show-note)
              ("'" . #'dired-annotator-show-note)))
```

Since this at least somehow works, and should display the files that
have been annotated via dired-annotator-show-icons, although, again this
does not work for me reliably for me.



-- 
Warning: Content may be disturbing to some audiences
I strongly condemn Putin's war of aggression against the Ukraine.
I support to deliver weapons to Ukraine's military. 
I support the ban of Russia from SWIFT.
I support the EU membership of the Ukraine. 
https://addons.thunderbird.net/en-US/thunderbird/addon/gmail-conversation-view/

**Comment by gunther-bachmann** [comment]
Yep. Send me a mail. In the meanwhile I will add some logging to the code, so that it will be easier to spot the problems ...

**Comment by ouboub** [comment]
solved with commit **bd6662ca8fa5**

**Comment by ouboub** [comment]
solved with commit **bd6662ca8fa5**
